### PR TITLE
Removed `package/meta.yml`

### LINKED ISSUE #1362: Remove our own-grown conda package
With conda packaging on conda-forge, we can probably remove `package` from our repo.

### PR DESCRIPTION
**Pull request by schlunma**
This PR removes  `package/meta.yml` as dicussed in #1522.

Closes #1362
Related to #1522

Documentation: https://esmvaltool--1540.org.readthedocs.build/projects/ESMValCore/en/1540/contributing.html#making-a-release

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [x] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🧪][2] and [🛠][1] [Documentation](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#documentation) is available
- [x] [🛠][1] [Unit tests](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#tests) have been added
- [x] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
- [x] [🛠][1] Any changed [dependencies have been added or removed](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#dependencies) correctly
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

**Comment by schlunma** [comment]
While unpinning `iris` in `meta.yml` I remembered that we wanted to remove this, that's how these two things ended up in the same PR.

It's basically two lines of code that change and one file that is removed. Is it really necessary to split this up? I also thought we already agreed on both changes?

**Comment by bouweandela** [comment]
Don't forget to update the documentation on making a release and the contribution guidelines

**Comment by schlunma** [comment]
> Don't forget to update the documentation on making a release and the contribution guidelines

Yes, was just about to do that. It makes sense to split the PR in that case.

I would need some input from someone who already manually published a conda package on [this](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/contributing.html#create-and-upload-the-conda-package), I have no idea how this needs to be modified.

**Comment by bouweandela** [comment]
> I would need some input from someone who already manually published a conda package on [this](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/contributing.html#create-and-upload-the-conda-package), I have no idea how this needs to be modified.

It should be replaced with instructions on how to update the conda-forge feedstock. We're no longer publishing to the esmvalgroup channel.

**Comment by bouweandela** [comment]
This section would also need an update: https://docs.esmvaltool.org/projects/esmvalcore/en/latest/contributing.html#dependencies

**Comment by schlunma** [comment]
I remove all appearances of `package/meta.yaml` and updated the release instructions accordingly.

**Review by valeriupredoi — APPROVED** [review]
looks good to me, cheers, Manu! Alas, I reckon we should be a bit more verbose about the instructions re: feedstock, but it's prob best to keep them short for now, since future release managers would need some coaching anyway - what you think @zklaus - briefer documentation and more on-the-job coaching or the other way round?

**Review by zklaus — APPROVED** [review]

**Comment by schlunma** [comment]
I'm merging this after @bouweandela had a look at this :+1:

File: doc/contributing.rst [hunk 1]
```diff
@@ -842,6 +807,34 @@ Follow these steps to create a new Python package:
 You can read more about this in
 `Packaging Python Projects <https://packaging.python.org/tutorials/packaging-projects/>`__.
 
+8. Create and upload the Conda package
```
Comment (bouweandela) [inline doc/contributing.rst:810]:
```suggestion
8. Create the Conda package
```

File: doc/contributing.rst [hunk 2]
```diff
@@ -740,8 +737,8 @@ to run their favourite recipe using this branch.
 3. Increase the version number
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-The version number is stored in ``esmvalcore/_version.py``,
-``package/meta.yaml``, ``CITATION.cff``. Make sure to update all files.
+The version number is stored in ``esmvalcore/_version.py``, ``CITATION.cff``.
+Make sure to update all files.
```
Comment (bouweandela) [inline doc/contributing.rst:741]:
```suggestion
Make sure to update both files.
```

**Review by bouweandela — APPROVED** [review]
Looks good, thanks for taking care of this!

**Pull request merged**